### ISSUE TITLE
Consolidate `switch%platform`, `browser_only` and `platform`

### ISSUE THREAD

**Issue by jchavarri** [ml-in-barcelona/server-reason-react#187]
There's now a bit confusing situation:

- `browser_only` extension
- `switch%platform` which include `Server` and `Client` branches
- `platform` which includes `js` and `native` options

We could use React namings and semantics to make things easier for folks to understand.

Maybe we could do something like this? 

- Remove `platform` as it's covering the same functionality as `browser_only`
- Rename `browser_only` to just `client`
- Add `server` to cover the `platform native` case
- Change `switch%platform` to `switch%environment`

So to summarize, we'd have:

- `switch%environment` with `Server` and `Client` branches
- `let%client`
- `let%server`

Wdyt?

**Comment by davesnx** [comment]
`let%client` and `let%server` can potentially collide with RSC ppx design thought. Personally, the `_only` suffix makes the intention clearer, so would rather keep `let%client_only` and `server_only`

**Comment by davesnx** [comment]
We totally agree, but we might not add `let%client` for RSC. 

If we *can* align both, it would be massive, but I'm afraid that isn't the case. Current browser_only discards functions and `use client` just changes react components.

We could make `use client` do all the work of browser_only (inside the module), but we can't remove browser_only since functions that aren't react components will also need it. The other interesting bit is that client components need the server implementation as well. which makes `use client` a bit to do.

**Comment by jchavarri** [comment]
> we can't remove browser_only since functions that aren't react components will also need it

to clarify, i don't suggest to remove it but rather rename it so there's no different "concepts" going around (e.g. "browser", "client").

**Comment by davesnx** [comment]
I like this proposal a lot, I have been thinking about it and now that we implemented `react.client.component` and react.async.component` I have a new view on this:

I can replace js and native for client/server, I would keep the _only for clarity and verbosity and move the "platform" name into environment.

It would look like:

```
[@platform js] → [@client_only]
[@platform native] → [@server_only]
let%browser_only → let%client_only
switch%platform → switch%environment
```

This reduces the cognitive load of having to remember different names for the same things. Unifies `[@platform js]` and `[@browser_only]` into one. Removes the need of payload in "@platform".

Any thoughts @pedrobslisboa @jchavarri @andreypopp?

**Comment by andreypopp** [comment]
> At first glance, the switch%environment name could be confusing 

Agree, maybe something like `switch%client_server`?
